### PR TITLE
DM-12611: FrameDict(FrameSet const &) broken

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: include/astshim/FrameDict.h [hunk 1]
```diff
@@ -105,8 +105,10 @@ class FrameDict : public FrameSet {
 
     @throws std::invalid_argument if two Frames in the FrameSet have the same non-empty domain.
     */
-    explicit FrameDict(FrameSet const &frameSet)
-            : FrameDict(reinterpret_cast<AstFrameSet *>(frameSet.copy()->getRawPtr())) {}
+    explicit FrameDict(FrameSet const &frameSet) : FrameSet(std::move(*frameSet.copy())), _domainIndexDict() {
+        _domainIndexDict = _makeNewDict(*this);
+    }
+
```
Comment (kfindeisen) [inline include/astshim/FrameDict.h:111]: Is the constructor from an `AstFrameSet*` still used for anything?

File: tests/test_frameDict.py [hunk 2]
```diff
@@ -60,12 +61,14 @@ def test_FrameDictOneFrameConstructor(self):
...
         frameSet = ast.FrameSet(self.frame1, self.zoomMap, self.frame2)
         frameDict = ast.FrameDict(frameSet)
-
         indata = np.array([[1.1, 2.1, 3.1], [1.2, 2.2, 3.2]])
         predictedOut = indata * self.zoom
         assert_allclose(frameDict.applyForward(indata), predictedOut)
         assert_allclose(frameDict.applyInverse(predictedOut), indata)
 
+        frameDict2 = makeFrameDict(frameSet)
+        self.assertEqual(frameDict2.getRefCount(), 1)
```
Comment (kfindeisen) [inline tests/test_frameDict.py:70]: Out of curiosity, what was the failure? Was `frameDict2.getRefCount()` 0?

**Review by kfindeisen — APPROVED** [review]
Looks good. Couple of minor comments.

File: tests/test_frameDict.py (hunk 2 above)
Comment (r-owen) [reply tests/test_frameDict.py:70]: Attempting to get the refcount fails.

File: include/astshim/FrameDict.h (hunk 1 above)
Comment (r-owen) [reply include/astshim/FrameDict.h:111]: Yes, it is used by Object::makeShim

**Pull request merged**